src/raftstore/store/snap.rs

```diff
@@ -1094,17 +1090,14 @@ mod v2 {
                     stat: &mut SnapshotStatistics)
                     -> RaftStoreResult<()> {
             if self.exists() {
-                match self.validate() {
-                    Ok(()) => return Ok(()),
-                    Err(e) => {
-                        warn!("[region {}] file {} is invalid, will rebuild: {:?}",
-                              region.get_id(),
-                              self.path(),
-                              e);
-                        self.delete();
-                        try!(self.init_for_building(snap));
-                    }
+                let r = self.validate();
+                if let Err(ref e) = r {
+                    warn!("[region {}] file {} is invalid, will rebuild: {:?}",
```

Comment:
If the snapshot is invalid (file is modified unexpectedly), this will not succeed until GC deletes the files, which may lead to panic.